core/system/src/mpu/vmpu_armv7m.c

```diff
@@ -704,45 +728,69 @@ void vmpu_arch_init_hw(void)
+     * |  box 0  |
+     * | public  |
+     * | memory  |
+     * +---------+ <-- uvisor_config.page_end: n/8th of _region_ size (not SRAM size)
+     * |  page   |
+     * |  heap   |
+     * +---------+ <-- aligned to page size
+     * | wastage | <-- wasted SRAM is less than 1 page size
+     * +---------+ <-- uvisor_config.page_start
+     * |  uVisor |
+     * |   bss   |
+     * '---------' <-- uvisor_config.sram_start, region start
+     *
+     * Example: The region size of a 96kB SRAM will be 128kB, and the page heap
+     *          end will have to be aligned to 16kB, _not_ 12kB (= 96kB / 8).
+     *
+     * Note: In case the uVisor bss section is moved to another memory region
+     *       (tightly-coupled memory for example), the page heap remains and
+     *       the same considerations apply. Therefore the uVisor bss section
+     *       location has no impact on this.
+     */
+    /* Calculate the region size by rounding up the SRAM size to the next power-of-two. */
+    const uint32_t total_size = (1 << vmpu_region_bits((uint32_t) __uvisor_config.sram_end - (uint32_t) __uvisor_config.sram_start));
+    /* The alignment is 1/8th of the region size = rounded up SRAM size. */
+    const uint32_t subregions_size = total_size / 8;
+    const uint32_t protected_size = (uint32_t) __uvisor_config.page_end - (uint32_t) __uvisor_config.sram_start;
+    /* The protected size must be aligned to the subregion size. */
+    if (protected_size % subregions_size != 0) {
         HALT_ERROR(SANITY_CHECK_FAILED,
-                   "The __uvisor_bss_end symbol (0x%08X)is not aligned to an MPU subregion boundary.",
-                   (uint32_t) __uvisor_config.bss_end);
+                   "The __uvisor_page_end symbol (0x%08X)is not aligned to an MPU subregion boundary.",
+                   (uint32_t) __uvisor_config.page_end);
     }
-    uint8_t subregions_mask = (uint8_t) ~(0xFFUL >> (8 - (original_size / subregions_size)));
+    /* Note: It's called the subregion _disable_ mask, so setting one bit in it _disables_ the
+     *       permissions in this subregion. Totally not confusing, amiright!? */
+    const uint8_t subregions_disable_mask = (uint8_t) ((1UL << (protected_size / subregions_size)) - 1UL);
 
-    /* Protect the uVisor SRAM.
-     * This region needs protection only if uVisor shares the SRAM with the
-     * host. The configuration requires that uVisor is right at the beginning of
-     * the SRAM. If not, uVisor would end up owning a memory regions that
-     * contains non-uVisor data. */
+    /* Unlock the upper SRAM subregion only. */
     vmpu_acl_static_region(
-        2,
-        (void *) SRAM_ORIGIN,
-        rounded_size,
-        UVISOR_TACL_SREAD | UVISOR_TACL_SWRITE | UVISOR_TACL_SUBREGIONS(subregions_mask)
+        1,
+        __uvisor_config.sram_start,
+        total_size,
+        UVISOR_TACLDEF_DATA | UVISOR_TACL_EXECUTE | UVISOR_TACL_SUBREGIONS(subregions_disable_mask)
```

Comment:
Yes, those were the permission from before. The user is allowed to execute code from RAM after all. Maybe we can turn it into `UVISOR_TACL_UEXECUTE`? I don't think we ever need to execute priviledged code from RAM.